src/main/scala/nl.knaw.dans.easy.solr4files/components/Solr.scala

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files.components
+
+import java.net.URL
+
+import nl.knaw.dans.easy.solr4files._
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.solr.client.solrj.SolrRequest.METHOD
+import org.apache.solr.client.solrj.impl.HttpSolrClient
+import org.apache.solr.client.solrj.request.ContentStreamUpdateRequest
+import org.apache.solr.client.solrj.response.UpdateResponse
+import org.apache.solr.client.solrj.{ SolrClient, SolrQuery }
+import org.apache.solr.common.util.{ ContentStreamBase, NamedList }
+
+import scala.language.postfixOps
+import scala.util.{ Failure, Success, Try }
+
+trait Solr extends DebugEnhancedLogging {
+  val solrUrl: URL
+  lazy val solrClient: SolrClient = new HttpSolrClient.Builder(solrUrl.toString).build()
+
+  def createDoc(item: FileItem, size: Long): Try[FileFeedback] = {
+    item.bag.fileUrl(item.path).flatMap { fileUrl =>
+      val solrDocId = s"${ item.bag.bagId }/${ item.path }"
+      val request = new ContentStreamUpdateRequest("/update/extract") {
+        setWaitSearcher(false)
+        setMethod(METHOD.POST)
+        addContentStream(new ContentStreamBase.URLStream(fileUrl))
+        setParam("literal.id", solrDocId)
+        setParam("literal.easy_file_size", size.toString)
+        for (
```

Comment:
replace this for-loop with

```scala
for ((key, value) <- item.bag.solrLiterals ++ item.ddm.solrLiterals ++ item.solrLiterals
     if value.trim.nonEmpty) {
  setParam(s"literal.easy_$key", value.replaceAll("\\s+", " ").trim)
}
```

This will on byte-code level transform the `if ...` into a `withFilter` and the body of the for-loop into a `foreach`.